### PR TITLE
verify: Exclude verify-external-dependencies from make verify

### PR DESCRIPTION
**Pull request by justaugustus**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

verify: Exclude verify-external-dependencies from `make verify`

Also:
- Adds verify-external-dependencies to `QUICK_PATTERNS`
- Renames `EXCLUDE_GODEP` to `EXCLUDE_DEPENDENCIES`
  (since Godeps has been removed from the repo)

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

/assign @cblecker
/hold until after https://github.com/kubernetes/kubernetes/pull/90963 has merged and has had some soak. Let's say until Monday, 5/18.

**Special notes for your reviewer**:

(Follow-up from a convo w/ @cblecker yesterday.)

tl;dr -- When we make changes to `build/dependencies.yaml`, `pull-kubernetes-verify` appears to not run `hack/verify-external-dependencies-version.sh` until much later in the job.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by justaugustus** [comment]
/area dependency
/priority important-soon

Not quite all SIG Release, but needs a SIG, so:
/sig release

File: hack/make-rules/verify.sh [hunk 1]
```diff
@@ -52,12 +52,12 @@ if [[ ${EXCLUDE_TYPECHECK:-} =~ ^[yY]$ ]]; then
     )
 fi
 
-
-# Exclude vendor checks in certain cases, if they're running in a separate job.
-if [[ ${EXCLUDE_GODEP:-} =~ ^[yY]$ ]]; then
+# Exclude dependency checks in certain cases, if they're running in a separate job.
+if [[ ${EXCLUDE_DEPENDENCIES:-} =~ ^[yY]$ ]]; then
```
Comment (cblecker) [inline hack/make-rules/verify.sh:56]: We can't change the variable name here, unless we update it throughout test-infra (and we would need to pick it backwards)

Comment (justaugustus) [reply hack/make-rules/verify.sh]: @cblecker -- reverted that change.

**Comment by justaugustus** [comment]
/hold cancel
(since https://github.com/kubernetes/kubernetes/pull/90963 has merged and it's 5/18 (when we said we'd lift the hold))
